slowfast/models/optimizer.py

```diff
@@ -55,18 +55,18 @@ def construct_optimizer(model, cfg):
     optim_params = [
         {
             "params": bn_parameters,
-            "weight_decay": cfg.BN.WEIGHT_DECAY,
+            "weight_decay": cfg.SOLVER.WEIGHT_DECAY if cfg.FSDP.ENABLED else cfg.BN.WEIGHT_DECAY,
             "apply_LARS": False,
         },
         {
             "params": non_bn_parameters,
             "weight_decay": cfg.SOLVER.WEIGHT_DECAY,
-            "apply_LARS": cfg.SOLVER.LARS_ON,
+            "apply_LARS": False if cfg.FSDP.ENABLED else cfg.SOLVER.LARS_ON,
```

Comment:
I feel like the code should crash instead of silently "correcting" arguments since it can lead to the users not knowing about these conversions and expecting different results.

Reply:
Resolved! Thanks.